grapevine/src/subscriber/aggregator.rs

```diff
@@ -0,0 +1,205 @@
+use std::sync::Arc;
+use futures::StreamExt;
+use tokio::sync::{mpsc, RwLockWriteGuard, watch};
+use tokio::sync::mpsc::error::TryRecvError;
+use tracing::{debug, error};
+
+use crate::{MessageSequencer, StateSync};
+use crate::subscriber::message::SubscriberStreamMessage;
+use crate::subscriber::sequencer::{SubscriberStream};
+
+fn get_delta<T>(data: SubscriberStream<T>) -> Option<T> {
+    if let SubscriberStream::Message { msg: SubscriberStreamMessage::Delta(delta), .. } = data {
+        Some(delta)
+    } else {
+        None
+    }
+}
+
+// aggregator and it's caller share same memory
+type State<S> = Arc<tokio::sync::RwLock<S>>;
+
+pub struct Aggregator<S: StateSync> {
+    state_rx: watch::Receiver<State<S>>,
+}
+
+impl<S: StateSync> Aggregator<S> {
+    pub fn new(
+        sequencer: MessageSequencer<S>,
+    ) -> Self {
+        let state = State::default();
+        let (state_tx, state_rx) = watch::channel(state.clone());
+        let (sequencer_tx, mut sequencer_rx) = mpsc::unbounded_channel();
```

Comment:
can we use bounded instead? can make configurable and just use a large default e.g. 10K

Reply:
make it bounded